adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/android/BrokerProxy.cs

```diff
@@ -417,9 +417,13 @@ private Bundle GetBrokerOptions(AuthenticationRequest request)
                 request.Resource);
             string computedRedirectUri = GetRedirectUriForBroker();
 
-            if (!string.IsNullOrEmpty(request.RedirectUri) && !string.Equals(computedRedirectUri, request.RedirectUri, StringComparison.OrdinalIgnoreCase))
+            if (string.IsNullOrWhiteSpace(request.RedirectUri))
```

Comment:
>string.IsNullOrWhiteSpace(request.RedirectUri) [](start = 16, length = 46)

@trwalke and I synced off-line. I don't think we need this check here. We already check for a null or empty redirectUri in the ValidateRedirectUri(), so is it necessary to check again here? Seems like we should fail fast and not get this far, so maybe failing when creating the broker payload is a better option if we don't already fail sooner? Then we won't need this line, or the new exception, and can just use line 424 w/the new improved error message. thoughts?

Reply:
cool...so i would do the fix as we discussed.

---
In reply to: [223531269](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1309#discussion_r223531269) [](ancestors = 223531269)